libs/stats/odc/stats/_tcw_percentiles.py

```diff
@@ -0,0 +1,95 @@
+"""
+Fractional Cover Percentiles
+"""
+from functools import partial
+from typing import Optional, Tuple, Dict
+import xarray as xr
+import numpy as np
+from odc.stats.model import Task
+from odc.algo.io import load_with_native_transform
+from odc.algo import keep_good_only
+from odc.algo._percentile import xr_percentile
+from odc.algo._masking import _xr_fuse, _fuse_mean_np
+from .model import StatsPluginInterface
+from . import _plugins
+
+
+NODATA = -9999 # output NODATA
+
+
+class StatsTCWPC(StatsPluginInterface):
+    
+    NAME = "ga_tcw_percentiles"
+    SHORT_NAME = NAME
+    VERSION = "0.0.1"
+    PRODUCT_FAMILY = "twc_percentiles"
+
+    def __init__(
+        self,
+        resampling: str = "bilinear",
+        coefficients: Dict[str, float] = {
+            "red": 0.0135, "blue": 0.2021, "green": 0.3102, "nir": 0.1584, "swir1": -0.6806, "swir2": -0.6109
+            },
+    ):
+        self.resampling = resampling
+        self.coefficients = coefficients
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        _measurments = ["tcw_pc_10", "tcw_pc_50", "tcw_pc_90"]
+        _measurments
+        return _measurments
+
+    def _native_tr(self, xx):
+        """
+        Loads data in its native projection.
+        """
+
+        bad = (xx["fmask"] & 0b0000_1101) != 0
```

Comment:
We need contiguity in the mask as well  bad:= `nbart_contiguity == 0`. Sorry I missed that in our conversation.
```
-   aliases:
    - nbart_contiguity
    dtype: uint8
    flags_definition:
        contiguous:
            bits:
            - 0
            values:
                '0': false
                '1': true
    name: oa_nbart_contiguity
    nodata: 255
    units: '1'
```

Reply:
I've added this in :) can you check that I've got the 1s and 0s right?